src/addons/Portal/Portal.js

```diff
@@ -0,0 +1,255 @@
+    closeOnDocumentClick: PropTypes.bool,
+
+    /** Controls whether or not the portal should close when escape is pressed is displayed. */
+    closeOnEscape: PropTypes.bool,
+
+    /** Controls whether or not the portal should close on blur of the trigger. */
+    closeOnTriggerBlur: PropTypes.bool,
+
+    /** Controls whether or not the portal should close when mousing out of the trigger. */
+    closeOnTriggerMouseLeave: PropTypes.bool,
+
+    /** Initial value of open. */
+    defaultOpen: PropTypes.bool,
+
+    /** Called when open changes from true to false */
+    onClose: PropTypes.func,
+
+    /** Called when open changes from false to true */
+    onOpen: PropTypes.func,
+
+    /** Controls whether or not the portal is displayed. */
+    open: PropTypes.bool,
+
+    /** Controls whether or not the portal should open when the trigger is clicked. */
+    openOnTriggerClick: PropTypes.bool,
+
+    /** Controls whether or not the portal should open on focus of the trigger. */
+    openOnTriggerFocus: PropTypes.bool,
+
+    /** Controls whether or not the portal should open when mousing over the trigger. */
+    openOnTriggerMouseOver: PropTypes.bool,
+
+    /** Element to be rendered in-place where the portal is defined. */
+    trigger: PropTypes.node,
+  }
+
+  static defaultProps = {
+    closeOnEscape: true,
+    closeOnDocumentClick: true,
+    openOnTriggerClick: true,
+  }
+
+  static autoControlledProps = [
+    'open',
+  ]
+
+  static _meta = _meta
+
+  componentDidMount() {
+    if (this.state.open) {
+      this.renderPortal()
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) { // eslint-disable-line complexity
+    // NOTE: Ideally the portal rendering would happen in the render() function
+    // but React gives a warning about not being pure and suggests doing it
+    // within this method.
+
+    // If the portal is open, render (or re-render) the portal and child.
+    if (this.state.open) {
+      this.renderPortal()
+    }
+
+    if (prevState.open && !this.state.open) {
+      debug('portal closed')
+      this.unmountPortal()
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmountPortal()
+  }
+
+  // ----------------------------------------
+  // Document Event Handlers
+  // ----------------------------------------
+
+  closeOnDocumentClick = (e) => {
+    if (!this.props.closeOnDocumentClick) return
+
+    debug('closeOnDocumentClick()')
+
+    e.stopPropagation()
+    this.close()
+  }
+
+  closeOnEscape = (e) => {
+    if (!this.props.closeOnEscape) return
+    if (keyboardKey.getCode(e) !== keyboardKey.Escape) return
+
+    debug('closeOnEscape()')
+
+    e.preventDefault()
+    this.close()
+  }
+
+  // ----------------------------------------
+  // Component Event Handlers
+  // ----------------------------------------
+
+  handleChildClick = (e) => {
+    debug('handleChildClick()')
+
+    // Prevent closeOnDocumentClick
+    e.nativeEvent.stopImmediatePropagation()
```

Comment:
Definitely, thanks for the reminder.  I'm on board with dropping findDOMNode.  You probably already know to use the callback form, see #405.